upgrade/Upgrade-2.4.1.php

```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
+                ['name' => 'sepa', 'color' => '#e3e1dc', 'config' => stripe_official::SEPA_DISPUTE],
+                ['name' => 'oxxo', 'color' => '#C23416', 'config' => stripe_official::OXXO_WAITING],
+            ];
+
+            $order_state_to_affect = array_map(
+                'cleanStripeOrderState',
+                array_column($stripe_order_states, 'name'),
+                array_column($stripe_order_states, 'color')
+            );
+
+            for ($i = 0; $i < 5; $i++) {
+                if (!$order_state_to_affect[$i]) {
+                    return false;
+                }
 
-            Configuration::updateValue(stripe_official::OS_SOFORT_WAITING, $result[0]['id_order_state']);
+                $order_state = new OrderState($order_state_to_affect[$i]);
+                $order_state->unremovable = true;
+                $order_state->module_name = $module->name;
+                $order_state->save();
+
+                Configuration::updateValue($stripe_order_states[$i]['config'], $order_state->id);
+            }
         }
 
         return true;
     } catch (PrestaShopDatabaseException $e) {
-        ProcessLoggerHandler::logError(
+        Stripe_officialClasslib\Extensions\ProcessLogger\ProcessLoggerHandler::logError(
             $e->getMessage(),
             null,
             null,
             'Upgrade 2.4.1'
         );
-        ProcessLoggerHandler::closeLogger();
+        Stripe_officialClasslib\Extensions\ProcessLogger\ProcessLoggerHandler::closeLogger();
         return false;
     } catch (PrestaShopException $e) {
-        ProcessLoggerHandler::logError(
+        Stripe_officialClasslib\Extensions\ProcessLogger\ProcessLoggerHandler::logError(
             $e->getMessage(),
             null,
             null,
             'Upgrade 2.4.1'
         );
-        ProcessLoggerHandler::closeLogger();
+        Stripe_officialClasslib\Extensions\ProcessLogger\ProcessLoggerHandler::closeLogger();
         return false;
     }
 }
+
+/**
+ * @throws PrestaShopDatabaseException
+ */
+function cleanStripeOrderState($name, $color)
+{
+    $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
+        'SELECT os.id_order_state FROM `' . _DB_PREFIX_ . 'order_state` os
+            INNER JOIN `' . _DB_PREFIX_ . 'order_state_lang` osl ON (os.id_order_state = osl.id_order_state)
+            WHERE osl.name LIKE "%' . $name . '%" AND os.color = "' . $color . '" GROUP BY os.id_order_state ORDER BY os.id_order_state DESC;');
+
+    $order_state_ids = array_column($result, 'id_order_state');
+
+    $order_state_id = $order_state_ids[0];
+
+    if (count($order_state_ids) > 1) {
+        unset($order_state_ids[0]);
+
+        $order_state_to_clean = implode(',', $order_state_ids);
+
+        if (!Db::getInstance(_PS_USE_SQL_SLAVE_)->execute('UPDATE `' . _DB_PREFIX_ . 'order_history` SET id_order_state = ' . $order_state_id . ' WHERE id_order_state IN (' . $order_state_to_clean . ');')
```

Comment:
You cannot do that. Probably other modules like ERP use PrestaShop state and history to manage synchronisation of order. you have no right to do that.